### ISSUE TITLE
Cannot update submodules

### ISSUE THREAD

**Issue by thomas-may-solidapps** [rowanj/gitx#274]
There seems to be no way to update submodules when I check out a new version in the main repository. This is why I´m still using the two-years-old Laullon fork, the feature exists there (right-click on submodule).

**Comment by dymv** [comment]
@thomas-may If I'm reading this right, double clicking on a submodule and updating it from a separate window could be helpful, isn't it?

**Comment by thomas-may-solidapps** [comment]
Not exactly. In version 0.14.95 I don´t see any way at all how I could update submodules. Submodules are shown in the menu, but there is no indication whether the version of the submodule has been updated or not. I can double-click on the submodule to open it, but I see no control for initializing or updating submodules.

There ist an old version 0.8.4 of GitX here: https://github.com/laullon/gitx
In that version I can click on a submodule with the right mouse button. This opens a popup menu with the options init, update, open, reveal in finder. Next to the submodules there are also icons which indicate whether the submodule is up to date or not.
To me this feature is a must-have. When I switch branches, I have to update submodules in most cases, and I need to see this.

I wonder if the feature could be taken into your fork from the old one. Or maybe I´m missing something?

**Comment by sven1904** [comment]
For me, this was the first missing feature that I mentioned directly after start using GitX-dev.
I was also using the previously mentioned GitX version 0.8.4 before, which supported that feature.
Always opening the submodule(s) and checking out the right commit after changing branches in the main module is way too cumbersome.
Of course, I can also open the terminal and update the submodules with `git submodule update`, but when I'm forced to use the console each time I change branches I don't need a GUI at all.
I'd appreciate the integration of that feature in one of the next releases.
Thanks in advance!

**Comment by muhqu** [comment]
@sven1904 you might want to check out https://github.com/gitx/gitx/ 
> GitX has a long history of various branches and versions maintained by various people over the years. This github org & repo are an attempt to consolidate and move forward with a current, common, community-maintained version. […]

**Comment by sven1904** [comment]
@muhqu thanks for the repo, but that versions also don't support the update feature (except versions 0.8.1-0.8.4).
I would just add the feature since I don't think that it takes that much time to implement.
But unfotrunately, I don't know anything about Objective-C.
Apperently, no other developers miss that feature. I just wonder why...

**Comment by muhqu** [comment]
@sven1904, you should file the issue [there](https://github.com/gitx/gitx/issues). It’s more likely to get picked up by someone....

> […] Apperently, no other developers miss that feature. I just wonder why...

I don’t miss that feature because I heavily use [DTerm](https://github.com/muhqu/dterm#dterm), which is a context sensitive popup-terminal. It allows me to execute any missing git command/alias/script from GitX by invoking a global shortcut to pull up the terminal. Depending on the current focused application window (GitX, VScode, Sublime, Eclipse...) the current working directory will be determined.
